plotter/boxplot_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"testing"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/vg"
+)
+
+func ExampleBoxPlot() {
+	// Create the sample data.
+	rand.Seed(0) // The default random seed is 1.
+	n := 100
+	uniform := make(ValueLabels, n)
+	normal := make(ValueLabels, n)
+	expon := make(ValueLabels, n)
+	for i := 0; i < n; i++ {
+		uniform[i].Value = rand.Float64()
+		uniform[i].Label = fmt.Sprintf("%4.4f", uniform[i].Value)
+		normal[i].Value = rand.NormFloat64()
+		normal[i].Label = fmt.Sprintf("%4.4f", normal[i].Value)
+		expon[i].Value = rand.ExpFloat64()
+		expon[i].Label = fmt.Sprintf("%4.4f", expon[i].Value)
+	}
+
+	// Make boxes for our data and add them to the plot.
+	uniBox, err := NewBoxPlot(vg.Points(20), 0, uniform)
+	if err != nil {
+		log.Fatal(err)
+	}
+	normBox, err := NewBoxPlot(vg.Points(20), 1, normal)
+	if err != nil {
+		log.Fatal(err)
+	}
+	expBox, err := NewBoxPlot(vg.Points(20), 2, expon)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Make a vertical box plot.
+	uniLabels, err := uniBox.OutsideLabels(uniform)
+	if err != nil {
+		log.Fatal(err)
+	}
+	normLabels, err := normBox.OutsideLabels(normal)
+	if err != nil {
+		log.Fatal(err)
+	}
+	expLabels, err := expBox.OutsideLabels(expon)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	p1, err := plot.New()
+	if err != nil {
+		log.Fatal(err)
+	}
+	p1.Title.Text = "Vertical Box Plot"
+	p1.Y.Label.Text = "plotter.Values"
+	p1.Add(uniBox, uniLabels, normBox, normLabels, expBox, expLabels)
+
+	// Set the X axis of the plot to nominal with
+	// the given names for x=0, x=1 and x=2.
+	p1.NominalX("Uniform\nDistribution", "Normal\nDistribution",
+		"Exponential\nDistribution")
+
+	err = p1.Save(200, 200, "examplePlots/verticalBoxPlot.png")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Now, make the same plot but horizontal.
```

Comment:
Since this is testing a different plot, can it please go into a different example function?